tools/preconf-rpc/rpcserver/rpcserver.go

```diff
@@ -187,44 +279,97 @@ func (s *JSONRPCServer) writeError(w http.ResponseWriter, id any, code int, mess
-		return
+	if resp.StatusCode != http.StatusOK {
+		return nil, resp.StatusCode, fmt.Errorf("proxy request returned status %d", resp.StatusCode)
 	}
-	if n == 0 {
-		http.Error(w, "Empty response from proxy", http.StatusInternalServerError)
-		return
+
+	out, err := io.ReadAll(io.LimitReader(resp.Body, defaultMaxBodySize))
+	if err != nil {
+		return nil, http.StatusInternalServerError, fmt.Errorf("failed to read proxy response: %w", err)
 	}
+
+	return out, resp.StatusCode, nil
 }
 
 func setCorsHeaders(w http.ResponseWriter) {
 	w.Header().Set("Access-Control-Allow-Origin", "*")
 	w.Header().Set("Access-Control-Allow-Methods", "POST, OPTIONS")
 	w.Header().Set("Access-Control-Allow-Headers", "Content-Type")
 }
+
+// short-circuit a few very common ERC-20 metadata calls
+// selectors: symbol 0x95d89b41, decimals 0x313ce567, name 0x06fdde03
+func maybeStubERC20Meta(method string, params []any) (stubbed bool, resp json.RawMessage) {
+	if method != "eth_call" || len(params) == 0 {
+		return false, nil
+	}
+	// parse minimal: [{"to":"0x..","data":"0x...."}, <blockTag?>]
+	callObj, ok := params[0].(map[string]any)
+	if !ok {
+		return false, nil
+	}
+	data, _ := callObj["data"].(string)
+	switch strings.ToLower(data) {
+	case "0x95d89b41": // symbol()
+		// return ABI-encoded string "TOKEN"
+		enc := "0x" +
+			"0000000000000000000000000000000000000000000000000000000000000020" + // offset
+			"0000000000000000000000000000000000000000000000000000000000000005" + // len
+			"544f4b454e000000000000000000000000000000000000000000000000000000" // "TOKEN"
+		return true, json.RawMessage(`"` + enc + `"`)
+	case "0x313ce567": // decimals()
+		enc := "0x" + "0000000000000000000000000000000000000000000000000000000000000012" // 18
+		return true, json.RawMessage(`"` + enc + `"`)
+	case "0x06fdde03": // name()
+		enc := "0x" +
+			"0000000000000000000000000000000000000000000000000000000000000020" +
+			"0000000000000000000000000000000000000000000000000000000000000005" +
+			"546f6b656e000000000000000000000000000000000000000000000000000000"
+		return true, json.RawMessage(`"` + enc + `"`)
+	default:
+		return false, nil
+	}
+}
+
+func pickTTL(method string, params json.RawMessage) time.Duration {
+	switch method {
+	case "net_version":
+		return 24 * time.Hour
+	case "eth_getCode":
+		return 24 * time.Hour
+	case "eth_getBlockByNumber":
+		// immutable if first param is a hex number (not "latest")
+		if strings.Contains(string(params), "\"0x") && !strings.Contains(string(params), "\"latest\"") {
+			return 24 * time.Hour
+		}
+		return 2 * time.Second
+	case "eth_feeHistory":
+		return 3 * time.Second
+	case "eth_call":
+		// if block tag provided and hex number → immutable
+		if strings.HasSuffix(string(params), "\"") { // cheap check
+			if strings.Contains(string(params), "\"0x") && !strings.Contains(string(params), "\"latest\"") {
+				return 24 * time.Hour
+			}
+		}
+		return 1 * time.Second
+	default:
```

Comment:
`eth_getBlockByNumber` with 'latest' would hit this default case of 2sec. Could it be feasible to just not cache calls like this that are time sensitive? As-is we could delay a client's knowledge of the latest L1 block by up to 2 sec